src/components/AllContents/index.tsx

```diff
@@ -0,0 +1,123 @@
+import React, { useEffect, useState } from 'react';
+import InputForm from '../InputForm';
+import YetList from '../YetList';
+import DoneList from '../DoneList';
+import { ITodoList } from 'interface';
+import getLocalValue from '../../hooks/getLocalValue';
+
+const Index = () => {
+  const listText = 'list';
+
+  const localData = getLocalValue();
+  // localDate 를 처음에 가져와서 새로고침해도 데이터가 유지되게 설정
+
+  const [yetNum, setYetNum] = useState(0);
+  const [doneNum, setDoneNum] = useState(0);
+  const [contents, setContents] = useState('');
+  const [list, setList] = useState(localData);
+
+  const handleContentsChange = (e: React.ChangeEvent<HTMLInputElement>) =>
+    setContents(e.target.value);
+  const handleFormSubmit = (e: React.SyntheticEvent) => {
+    e.preventDefault();
+    const val = getLocalValue();
+    const checkArry = val.filter((item: ITodoList) => {
+      return item.contents === contents;
+    });
+    if (checkArry.length === 0) {
+      const obj = {
+        contents,
+        id: Date.now(),
+        isDone: false,
+      };
+      setList((prev: []) => [...prev, obj]);
+      setContents('');
+    }
+  };
+  const onToggle = (e: React.MouseEvent<HTMLButtonElement>) => {
+    // 클릭된 값
+
+    const text = (e.target as HTMLElement).textContent;
+    // 로컬스토리지에 저장된 값
+    const data = getLocalValue();
+    // 두 값을 이용해서 클릭된 값만 isDone 상태 토글
+    let indexClicked = 0;
+    let obj;
+    data.map((item: ITodoList, index: number) => {
+      if (item.contents === text) {
+        obj = { ...item };
+        obj.isDone = !item.isDone;
+        indexClicked = index;
+      }
+    });
+    const updatedDate = [...data];
+    updatedDate[indexClicked] = obj;
+    setList(updatedDate);
```

Comment:
이런 식으로도 단순화 할 수 있을 것 같아요!
```suggestion
    setList([...data.map((item: ITodoList):ITodoList=>(
      item.contents === text ? {...item, isDone: !item.isDone} : item
    ))])
```

Reply:
오 감사합니다! 제가 대헌님 코드에 남겼던 코드도 그렇게 간단하게 줄여서 사용할 수 있는 거 보도 놀랐어요. 이 부분도 좋은 것 같네요 ㅎㅎ 이렇게 고쳐보도록 해봐야겠어요